src/backend/distributed/transaction/distributed_deadlock_detection.c

```diff
@@ -230,25 +235,28 @@ CheckForDistributedDeadlocks(void)
  */
 static bool
 CheckDeadlockForTransactionNode(TransactionNode *startingTransactionNode,
-								TransactionNode **transactionNodeStack,
+								int maxStackDepth,
 								List **deadlockPath)
 {
 	List *toBeVisitedNodes = NIL;
-	int currentStackDepth = 0;
+	const int rootStackDepth = 0;
+	TransactionNode **transactionNodeStack =
+		palloc0(maxStackDepth * sizeof(TransactionNode *));
 
 	/*
 	 * We keep transactionNodeStack to keep track of the deadlock paths. At this point,
 	 * adjust the depth of the starting node and set the stack's first element with
 	 * the starting node.
 	 */
-	transactionNodeStack[currentStackDepth] = startingTransactionNode;
+	transactionNodeStack[rootStackDepth] = startingTransactionNode;
 
-	PrependOutgoingNodesToQueue(startingTransactionNode, currentStackDepth,
+	PrependOutgoingNodesToQueue(startingTransactionNode, rootStackDepth,
 								&toBeVisitedNodes);
 
 	/* traverse the graph and search for the deadlocks */
 	while (toBeVisitedNodes != NIL)
 	{
+		int currentStackDepth;
```

Comment:
So we have agreed upon not initializing variables?

Reply:
😈

Reply:
If you want to push back on this I'm down to talk about it but it's a silly convention and I'm willing to block progress until we either stop enforcing initialization to always happen or stop requiring that variables are only declared at the beginning of a scope. This is a 10 minute conversation after which we decide to stop doing the wrong thing for, and I use this word carefully, _literally_ zero cost.

Reply:
I'm 100% on board with the convention change and have noticed you sneaking it in inside various patches. I agree the benefits of not initializing are real and I don't know when the benefits of initializing actually exist (at least given our compiler flags, which will catch failing to initialize).